docs/src/install.rst

```diff
@@ -48,6 +48,12 @@ Build Requirements
 - `cmake <https://cmake.org/>`_ -- version 3.24 or later, and ``make``
 - Xcode >= 14.3 (Xcode >= 15.0 for macOS 14 and above)
 
+.. note::
+    Ensure your shell environment and tools are running native ``arm64``,
+    not ``x86_64`` via Rosetta. Verify the shell by confirming ``uname -m``
+    outputs ``arm64``. If installed via homebrew, verify ``which cmake``
+    begins with ``/opt/homebrew/``.
```

Comment:
Awesome, thanks for adding it! 

What happens now if you build with x86? Does it break in or print an error message? 

If so would be great to add a line to the trouble shooting section below along the lines of "If you see this error when running the tests, then here is the fix""

Also it would be helpful to include a comment (probably in the trouble shooting section) about how to fix it. So for the shell, reinstall presumably? For cmake install a new cmake with brew and use that or set your PATH appropriately.

Reply:
It builds fine with no errors, just this warning at the beginning of the cmake output:
```
-- Building MLX for x86_64 processor on Darwin
CMake Warning at CMakeLists.txt:33 (message):
  Building for x86_64 on macOS is not supported.  If you are on an Apple
  silicon system, make sure you are building for arm64.
```
Wonder if cmake should just fail if building for non-arm64 unless/until these issues are fixed. Not sure if cross-platform is a goal given the nature of the project.

Sure, I can add some info on what I did to fix it in the troubleshooting section 👍

Reply:
> Not sure if cross-platform is a goal given the nature of the project.

It's a good point but merits a longer discussion. For now I think updating the docs is a great place to start.

Reply:
Agreed, updated PR with troubleshooting details.